spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+        {
+            // [START add_column]
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string alterStatement =
+                $"ALTER TABLE Albums ADD COLUMN MarketingBudget INT64";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.UpdateDatabaseDdlAsync(
+                    new DatabaseName(projectId, instanceId, databaseId),
+                new[] { alterStatement });
+            // Poll until the returned long-running operation is complete.
+            var updateDatabaseResponse =
+                await createDatabaseRequest.PollUntilCompletedAsync();
+            Console.WriteLine("Added the MarketingBudget column.");
+            return updateDatabaseResponse.Result;
+            // [END add_column]
+        }
+
+        public static async Task QuerySampleDataAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/"
+            + $"{instanceId}/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "
+                        + $"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]}");
+                    }
+                }
+            }
+            return;
+            // [END query_data]
+        }
+
+
+        public static async Task QueryDataWithIndexAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data_with_index]
+            // [START read_data_with_index]
+            string startTitle = "Aardvark";
+            string endTitle = "Goo";
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT AlbumId, AlbumTitle, MarketingBudget FROM Albums@ "
+                    + "{FORCE_INDEX=AlbumsByAlbumTitle} "

```

Comment:
we should get someone from spanner to comment  -- but i would assume it would do the correct thing and use that index if you used that column in orderby... yes?